From 6d4f48ee9c10b70d04c988e41e9d6cf9e11d7210 Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Fri, 14 Sep 2018 14:52:38 +0530
Subject: [PATCH 0397/1852] drm: xlnx: sdi: add video mode hdisplay, flags to
 the shared data

commit 5e3933eee16f635d51ef6d6dbba578dcae0e69f6 from
https://github.com/Xilinx/linux-xlnx.git

drm video mode flags carry information of video, whether it is in
interlaced mode or not. hdisplay conveys horizontal resolution.
This will be used by SDI audio embed block.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_sdi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_sdi.c b/drivers/gpu/drm/xlnx/xlnx_sdi.c
index 1aebc7d4581c..8ce0a60090a6 100644
--- a/drivers/gpu/drm/xlnx/xlnx_sdi.c
+++ b/drivers/gpu/drm/xlnx/xlnx_sdi.c
@@ -867,7 +867,9 @@ static void xlnx_sdi_encoder_atomic_mode_set(struct drm_encoder *encoder,
 
 	/* parameters for sdi audio */
 	sdi->video_mode.vdisplay = adjusted_mode->vdisplay;
+	sdi->video_mode.hdisplay = adjusted_mode->hdisplay;
 	sdi->video_mode.vrefresh = adjusted_mode->vrefresh;
+	sdi->video_mode.flags = adjusted_mode->flags;
 
 	xlnx_stc_sig(sdi->base, &vm);
 }
-- 
2.31.1

